test/test_memory.cpp

```diff
@@ -172,20 +172,109 @@ TEST_F(MemoryTest, RefineCachePolicy) {
 }
 
 class TMATest : public NVFuserTest {
+ protected:
   void SetUp() override {
-    // requires Hopper or newer
-    if (!deviceMajorMinorCheck(9)) {
+    if (cudaArchGuardShouldSkip(9, 0)) {
       GTEST_SKIP() << "skipping tests on pre-Hopper GPUs";
     }
     NVFuserTest::SetUp();
   }
 };
 
-TEST_F(TMATest, LoadCompleteTensor1D) {
+// Check that there is an xor "^" somewhere in the kernel
+class XorFinder : private kir::IrVisitor {
+ private:
+  using kir::IrVisitor::dispatch;
+  using kir::IrVisitor::handle;
+  bool found = false;
+
+  // We recursively goes into val's definition and its inputs and outputs, this
+  // is used to prevent infinite recursion
+  std::unordered_set<Expr*> visited;
+
+  void handle(kir::TensorIndex* ti) final {
+    handle(ti->index());
+  }
+
+  void handle(Val* v) {
+    if (v->definition() != nullptr) {
+      dispatch(v->definition());
+    }
+  }
+
+  void dispatch(Expr* expr) final {
+    if (found || !visited.insert(expr).second) {
+      return;
+    }
+    if (expr->isA<kir::ForLoop>() || expr->isA<kir::IfThenElse>()) {
+      kir::IrVisitor::dispatch(expr);
+      return;
+    }
+    if (expr->isA<BinaryOp>()) {
+      auto bin_op = expr->as<BinaryOp>();
+      if (bin_op->getBinaryOpType() == BinaryOpType::BitwiseXor) {
+        found = true;
+        return;
+      }
+    }
+    for (auto val : expr->inputs()) {
+      dispatch(val);
+    }
+    for (auto val : expr->outputs()) {
+      dispatch(val);
+    }
+  }
+
+ public:
+  static bool findXor(kir::Kernel* kernel) {
+    XorFinder finder;
+    finder.handle(kernel->topLevelExprs());
+    return finder.found;
+  }
+};
+
+using TMATestParams = std::tuple<MmaInputSmemSwizzle, DataType>;
+
+class TMALdstTest : public TMATest,
+                    public ::testing::WithParamInterface<TMATestParams> {
+ protected:
+  MmaInputSmemSwizzle swizzle;
+  DataType dtype;
+
+  int64_t innerDimSize() const {
+    return getBytesFromSwizzle(swizzle) / dataTypeSize(dtype);
+  }
+
+  int64_t swizzleSize() const {
+    return getBytesFromSwizzle(swizzle) / 16;
+  }
+
+  void SetUp() override {
+    TMATest::SetUp();
+    swizzle = std::get<0>(GetParam());
+    dtype = std::get<1>(GetParam());
+  }
+};
+
+// Assuming tv is currently scheduled as a 1D flat tensor, schedule the TMA
+// swizzle for it.
+void scheduleTMASwizzle(TensorView* tv, int64_t swizzle_size) {
```

Comment:
For now, this is just a function in this test file. In the future, I will refactor to make it a scheduler utility function, and this utility function will be used to schedule mma inputs.

Reply:
A minor suggestion, instead of using literal/magic number (`16`) maybe we should introduce a constant?

Reply:
Fixed in https://github.com/NVIDIA/Fuser/pull/1818